internal/sysfs/sysfs.go

```diff
@@ -345,20 +349,19 @@ func unwrapOSError(err error) error {
 		return syscall.ENOENT
 	case fs.ErrClosed:
 		return syscall.EBADF
-	case os.ErrInvalid:
-		return syscall.EINVAL
-	case os.ErrExist:
-		return syscall.EEXIST
-	default:
-		if errors.Is(err, os.ErrExist) {
-			return syscall.EEXIST
-		}
-		if errors.Is(err, os.ErrNotExist) {
-			return syscall.ENOENT
-		}
-		if errors.Is(err, os.ErrPermission) {
-			return syscall.EPERM
-		}
+	}
+	return syscall.EIO
+}
+
+// underlyingError returns the underlying error if a well-known OS error type.
+func underlyingError(err error) error {
```

Comment:
just not sure about the naming here because it's a no-op if there is no underlying error... something like "`unwrapIfOSError`"? (is that an acceptable name in Go?)

Reply:
oh heh this is literally the name used in something similar in go. I sometimes choose exact same names, but in this case forgot to comment. lemme